aiosmtpd/tests/test_server.py

```diff
@@ -57,6 +67,30 @@ def in_wsl():
     return "microsoft" in platform.release().casefold()
 
 
+def in_cygwin():
+    return platform.system().casefold().startswith("cygwin")
+
+
+@pytest.fixture(scope="module")
+def safe_socket_dir() -> Generator[Path, None, None]:
+    # See:
+    #   - https://github.com/aio-libs/aiohttp/issues/3572
+    #   - https://github.com/aio-libs/aiohttp/pull/3832/files
+    #   - https://unix.stackexchange.com/a/367012/5589
+    tmpdir = Path(mkdtemp()).absolute()
+    assert len(str(tmpdir)) <= 87  # 92 (max on HP-UX) minus 5 (allow 4-char fn)
+    #
+    yield tmpdir
+    #
+    plist = [p for p in tmpdir.rglob("*")]
+    for p in reversed(plist):
+        if p.is_dir():
+            p.rmdir()
+        else:
+            p.unlink()
+    tmpdir.rmdir()
```

Comment:
To create socket successfully in MacOS. The GA runner path is too deep on MacOS.